includes/Core/Util/Activation.php

```diff
@@ -155,17 +155,84 @@ private function get_activation_notice() {
 					ob_start();
 					?>
 					<script type="text/javascript">
-					if( 'undefined' !== typeof sendAnalyticsTrackingEvent ) {
-						sendAnalyticsTrackingEvent( 'plugin_setup', 'plugin_activated' );
-					}
+						document.addEventListener( 'DOMContentLoaded' , function() {
+							if ( 'undefined' !== typeof sendAnalyticsTrackingEvent ) {
+								sendAnalyticsTrackingEvent( 'plugin_setup', 'plugin_activated' );
+							}
+
+							var trackingScriptPresent = !! googlesitekit.admin.trackingOptIn;
+
+							if ( googlesitekit.admin.trackingOptIn ) {
+								document.getElementById( 'googlesitekit-opt-in' ).checked = googlesitekit.admin.trackingOptIn;
```

Comment:
Element lookups like this should be coded defensively, in case the element does not exist. We've had several issues with that in the past, and even if it's highly unlikely here that those elements do not exist, let's make sure this doesn't cause JS errors.

I'd also recommend gathering the elements _once_ at the beginning of the function in `var`s and then referencing those variables. This improves readability of the code and prevents duplicate lookups.